File: week2/readability/readability.c
#include <stdio.h>

int is_alphabetical(char letter);
int is_blank(char letter);
int is_punctuation(char letter);

int main(void)
{
    char text[1024];
    printf("Text: ");
    fgets(text, 1024, stdin);
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        letters += is_alphabetical(text[i]) ? 1 : 0;
        words += is_blank(text[i]);
        sentences += is_punctuation(text[i]);
    }

    float average_letters = ((float) letters / words) * 100;
    float average_sentences = ((float) sentences / words) * 100;
    int index = (0.0588 * average_letters - 0.296 * average_sentences - 15.8) + 0.5;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

    return 0;
}

int is_alphabetical(char letter)
{
    return ((letter >= 'a' && letter <= 'z') || (letter >= 'A' && letter <= 'Z')) ? 1 : 0;
}

int is_blank(char letter)
{
    return (letter == ' ') ? 1 : 0;
}

int is_punctuation(char letter)
{
    return (letter == '.' || letter == '?' || letter == '!') ? 1 : 0;
}

File: week6/sentimental-readability/readability.py
def main():
    text = input("Text: ")
    letters = 0
    words = 1
    sentences = 0

    for char in text:
        letters += 1 if is_alphabetical(char) else 0
        words += is_blank(char)
        sentences += is_punctuation(char)

    average_letters = (letters / words) * 100
    average_sentences = (sentences / words) * 100
    index = int(0.0588 * average_letters - 0.296 *
                average_sentences - 15.8 + 0.5)

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


def is_alphabetical(letter):
    return 1 if ('a' <= letter <= 'z' or 'A' <= letter <= 'Z') else 0


def is_blank(letter):
    return 1 if letter == ' ' else 0


def is_punctuation(letter):
    return 1 if letter in '.?!' else 0


main()

